opsimulator/opsimulator.go

```diff
@@ -148,8 +152,9 @@ func (opSim *OpSimulator) startBackgroundTasks() {
 	// Relay deposit tx from L1 to L2
 	opSim.bgTasks.Go(func() error {
 		depositTxCh := make(chan *types.DepositTx)
+		l1DepositTxnLogCh := make(chan *types.Log)
 		portalAddress := common.Address(opSim.Config().L2Config.L1Addresses.OptimismPortalProxy)
-		sub, err := SubscribeDepositTx(context.Background(), opSim.l1Chain.EthClient(), portalAddress, depositTxCh)
+		sub, err := SubscribeDepositTx(context.Background(), opSim.l1Chain.EthClient(), portalAddress, depositTxCh, l1DepositTxnLogCh)
```

Comment:
similar to the indexer/relayer design for l2 to l2 messages, it will be more maintainable to 

have the indexer have a start function and run separately from the opsimulator. the op simulator then "listens" to the indexer to get the stream of deposit tx

Reply:
while indexer listens for new tranction a transaction event should also be sent from indexer to opsim to SendTransaction to done while the indexer store ref to the initial depTxn sub message

Reply:
or another way could be both opsim and indexer can both sub to deposit txn, the op sim executes transaction while the indexer just stores ref?